### PR TITLE
Feature/extend orm client

### PR DESCRIPTION
**Pull request by jonkan81**
Adds:
* Parametrized ORM queries
* Parametrized execute statements
* Interfaces for RqliteClient and RqliteOrmClient useful for dependency injection, unit testing etc.

### CODE REVIEW

**Comment by otoolep** [comment]
@arthrp -- would you review this, and merge if it looks good to you?

**Comment by otoolep** [comment]
Thanks for your contribution, @jonkan81

**Comment by otoolep** [comment]
Say we want to enable continuous testing, what's a good environment? Something on CircleCI?

**Comment by arthrp** [comment]
@otoolep 
Sorry, due to personal issues I'm not too active here. For CI the easiest option is Github actions

File: RqliteDotnet/RqliteClient.cs [hunk 1]
```diff
@@ -1,12 +1,61 @@
...
+using System.Net.Http;
 using System.Text;
 using System.Text.Json;
-using RqliteDotnet.Dto;
+using System.Threading.Tasks;
 
 namespace RqliteDotnet;
 
-public class RqliteClient
+public interface IRqliteClient
```
Comment (arthrp) [inline RqliteDotnet/RqliteClient.cs]: Please extract `IRqliteClient` to a separate file for clarity

File: RqliteDotnet/HttpClientExtensions.cs [hunk 2]
```diff
@@ -4,12 +4,15 @@ namespace RqliteDotnet;
 
 public static class HttpClientExtensions
 {
-    public static async Task<T> SendTyped<T>(this HttpClient client, HttpRequestMessage request)
+    public static async Task<T?> SendTyped<T>(this HttpClient client, HttpRequestMessage request)
```
Comment (arthrp) [inline RqliteDotnet/HttpClientExtensions.cs]: Not sure it can return null here?

**Comment by arthrp** [comment]
@SjoVJonatan Thanks for your contribution! There're still few outstanding minor issues but I think we can merge now

**Pull request merged**